src/modules/api/accessControl/services/index.ts

```diff
@@ -0,0 +1,115 @@
+import { PrismaService } from "@/modules/core/prisma/services";
+import { Prisma } from "@prisma/client";
+import { CreateRoleDto, FetchRolesDto } from "../dtos";
+import { buildResponse } from "@/utils";
+import { HttpStatus, Injectable } from "@nestjs/common";
+import slugify from "slugify";
+import { DuplicateRoleException, RoleNotFoundException } from "../errors";
+
+@Injectable()
+export class AccessControlService {
+    constructor(private prisma: PrismaService) {}
+
+    async fetchRoles(options: FetchRolesDto) {
+        const queryOptions: Prisma.RoleFindManyArgs = {
+            where: {},
+            select: {
+                name: true,
+            },
+        };
+
+        if (options.roleName) {
+            queryOptions.where.name = { search: options.roleName };
+        }
+
+        const roles = await this.prisma.role.findMany(queryOptions);
+
+        return buildResponse({
+            message: "Roles retrieved successfully",
+            data: roles,
+        });
+    }
+
+    private async validatePermission(permission: Array<number>) {
+        const permissions = await this.prisma.permission.findMany({
```

Comment:
consider using a more effective algorithm. The idea is to make sure that permission IDs coming from the frontend exist in the db.